subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/jvm/JavaToolchainBuildOperationsFixture.groovy

```diff
@@ -49,7 +71,12 @@ class JavaToolchainBuildOperationsHelper {
         ]
     }
 
-    static List<BuildOperationRecord.Progress> toolchainEvents(BuildOperationsFixture operations, String taskPath) {
+    static List<BuildOperationRecord.Progress> filterByJavaVersion(List<BuildOperationRecord.Progress> events, JvmInstallationMetadata jdkMetadata) {
+        events.findAll { it.details.toolchain.javaVersion == jdkMetadata.javaVersion }
+    }
+
+    List<BuildOperationRecord.Progress> toolchainEvents(String taskPath) {
+        initIfNeeded()
```

Comment:
Is there any class which uses this method in `setup`? Since this is the only call to `initIfNeeded` I suppose you can remove the method again. Or you'd add a public getter for the operation which would init it if needed. WDYT?

Reply:
I added this similarly to how it was in `DirectoryBuildCacheFixture` to make a bit more future proof for the users of this new trait. But the tests using this new trait don't use it in the setup, so this `initIfNeeded` can be removed.

I am not sure we should expose the `operations` until there is a reason for it. Maybe better to extend the trait further if it is needed in the future.